doc/source/config.rst

```diff
@@ -3,28 +3,88 @@
 
-Environment variables (typically not necessary):
+Some configuration is possible:
 
-- ``PRISMS_JOBS_DB``: 
+- ``PRISMS_JOBS_DIR``: (optional, default=``$HOME/.prisms_jobs``) 
 
-    The SQLite jobs database is stored by default at ``$HOME/.prisms_jobs/jobs.db``. 
-    If ``PRISMS_JOBS_DB`` is set, then the jobs database is stored at 
-    ``$PRISMS_JOBS_DB/jobs.db``.
+    The jobs database is stored at ``$PBS_JOB_DIR/jobs.db``.
 
-- ``PRISMS_JOBS_SOFTWARE``: 
+- ``PRISMS_JOBS_DIR/config.json``: 
 
-    By default, ``prisms-jobs`` will attempt to automatically 
-    detect ``'torque'`` (by checking for the 'qstat' executable) or ``'slurm'`` (by 
-    checking for the 'sbatch' executable). The ``'default'`` module provides stubs to 
-    enable testing/use on systems with no job management software. If ``PRISMS_JOBS_SOFTWARE``
-    is set to any other value, it is treated as the name of a Python module containing 
-    a custom interface which ``prisms_jobs`` will attempt to import and use.
+    Automatically generated JSON configuration file storing settings:
+    
+    - ``"dbpath"``: (str) 
+    
+        The location of the SQLite jobs database.
+    
+    - ``"software"``: (str) 
+    
+        The job submission software interface to use. ``"torque"`` or ``"slurm"``
+        is automatically detected if present.
+        
+        +-------------------+------------------------------------------------+
+        |"torque"           |TORQUE                                          |
+        +-------------------+------------------------------------------------+
+        |"slurm"            |Slurm                                           |
+        +-------------------+------------------------------------------------+
+        |"default" (or null)|Empty stub, does nothing                        |
+        +-------------------+------------------------------------------------+
+        |other              |The name of an existing findable python module  |
+        |                   |implementing an interface                       |
+        +-------------------+------------------------------------------------+
 
-- ``PRISMS_JOBS_UPDATE``: 
+    - ``"write_submit_script"``: (bool, optional, default=false) 
+    
+        If ``true``, submit jobs by first writing a submit script file and then 
+        submitting it. Otherwise, by default, the job is submitted via the command 
+        line.
+    
+    - ``"update_method"``: (str, optional, default="default")
+        
+        Controls which jobs are updated when JobDB.update() is called.
+        
+        +-------------------+------------------------------------------------+
+        |"default" (or null)| Select jobs with jobstatus != 'C'              |
+        +-------------------+------------------------------------------------+
+        |"check_hostname"   | Select jobs with jobstatus != 'C' and matching |
+        |                   | hostname                                       |
```

Comment:
We should probably mention the use case for "check_hostname": compute clusters where (multiple machines with) different queues share the same PRISMS_JOBS_DIR (e.g., the same homedir).